core/src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -237,7 +239,7 @@ private void extract(PluginHandle pluginHandle, Terminal terminal, Path pluginFi
         Path root = tmp.resolve(pluginHandle.name);
         unzipPlugin(pluginFile, root);
 
-        // find the actual root (in case its unzipped with extra directory wrapping)
+        // find the actual root for the elasticsearch plugin (the zip may contain other system plugins).
         root = findPluginRoot(root);
```

Comment:
We should not make these changes until we remove site plugin support. This part right here, does not play well with this change at all.